File: backend/internal/model/learning_asset.go
package model

import "time"

type LearningAssetJob struct {
	ID              string     `db:"id" json:"id"`
	UserID          string     `db:"user_id" json:"-"`
	LearningSpaceID string     `db:"learning_space_id" json:"learning_space_id"`
	JobType         string     `db:"job_type" json:"job_type"`
	Status          string     `db:"status" json:"status"`
	Progress        int        `db:"progress" json:"progress"`
	Attempts        int        `db:"attempts" json:"attempts"`
	MaxAttempts     int        `db:"max_attempts" json:"max_attempts"`
	AvailableAt     time.Time  `db:"available_at" json:"-"`
	StartedAt       *time.Time `db:"started_at" json:"started_at"`
	CompletedAt     *time.Time `db:"completed_at" json:"completed_at"`
	ErrorCode       *string    `db:"error_code" json:"error_code"`
	ErrorMessage    *string    `db:"error_message" json:"error_message"`
	CreatedAt       time.Time  `db:"created_at" json:"created_at"`
	UpdatedAt       time.Time  `db:"updated_at" json:"updated_at"`
}

type LearningNode struct {
	ID                string   `db:"id" json:"id"`
	LearningSpaceID   string   `db:"learning_space_id" json:"learning_space_id"`
	Name              string   `db:"name" json:"name"`
	NodeType          string   `db:"node_type" json:"node_type"`
	Description       string   `db:"description" json:"description"`
	ExamWeight        float64  `db:"exam_weight" json:"exam_weight"`
	EstimatedMinutes  int      `db:"estimated_minutes" json:"estimated_minutes"`
	SourceChunkID     *string  `db:"source_chunk_id" json:"source_chunk_id"`
	SortOrder         int      `db:"sort_order" json:"sort_order"`
	PositionX         *float64 `db:"position_x" json:"position_x"`
	PositionY         *float64 `db:"position_y" json:"position_y"`
	UserEdited        bool     `db:"user_edited" json:"user_edited"`
	MasteryScore      float64  `db:"mastery_score" json:"mastery_score"`
	MasteryStatus     string   `db:"mastery_status" json:"mastery_status"`
	EvidenceCount     int      `db:"evidence_count" json:"evidence_count"`
	MasteryConfidence float64  `db:"mastery_confidence" json:"mastery_confidence"`
}
type LearningEdge struct {
	ID           string `db:"id" json:"id"`
	FromNodeID   string `db:"from_node_id" json:"from_node_id"`
	ToNodeID     string `db:"to_node_id" json:"to_node_id"`
	RelationType string `db:"relation_type" json:"relation_type"`
	UserEdited   bool   `db:"user_edited" json:"user_edited"`
}
type KnowledgeArticle struct {
	ID            string  `db:"id" json:"id"`
	NodeID        string  `db:"node_id" json:"node_id"`
	Title         string  `db:"title" json:"title"`
	Body          string  `db:"body" json:"body"`
	SourceChunkID *string `db:"source_chunk_id" json:"source_chunk_id"`
	SortOrder     int     `db:"sort_order" json:"sort_order"`
	UserEdited    bool    `db:"user_edited" json:"user_edited"`
}
type LearningPlan struct {
	ID               string    `db:"id" json:"id"`
	PlanDate         string    `db:"plan_date" json:"plan_date"`
	Title            string    `db:"title" json:"title"`
	TotalMinutes     int       `db:"total_minutes" json:"total_minutes"`
	GenerationReason string    `db:"generation_reason" json:"generation_reason"`
	CreatedAt        time.Time `db:"created_at" json:"created_at"`
}
type PlanStage struct {
	ID            string `db:"id" json:"id"`
	PlanID        string `db:"plan_id" json:"plan_id"`
	FocusNodeID   string `db:"focus_node_id" json:"focus_node_id"`
	Title         string `db:"title" json:"title"`
	Description   string `db:"description" json:"description"`
	Status        string `db:"stage_status" json:"status"`
	EstimatedDays int    `db:"estimated_days" json:"estimated_days"`
	SortOrder     int    `db:"sort_order" json:"sort_order"`
}
type PlanNode struct {
	ID               string `db:"id" json:"id"`
	PlanID           string `db:"plan_id" json:"plan_id"`
	NodeID           string `db:"node_id" json:"node_id"`
	TaskType         string `db:"task_type" json:"task_type"`
	Title            string `db:"title" json:"title"`
	EstimatedMinutes int    `db:"estimated_minutes" json:"estimated_minutes"`
	SortOrder        int    `db:"sort_order" json:"sort_order"`
}

File: frontend/src/types/learning-asset.ts
export type AssetJobStatus = 'queued' | 'processing' | 'succeeded' | 'failed'
export interface LearningAssetJob { id:string; learning_space_id:string; job_type:'knowledge_assets'|'learning_plan'; status:AssetJobStatus; progress:number; attempts:number; max_attempts:number; started_at:string|null; completed_at:string|null; error_code:string|null; error_message:string|null; created_at:string; updated_at:string }
export interface LearningNode { id:string; learning_space_id:string; name:string; node_type:string; description:string; exam_weight:number; estimated_minutes:number; source_chunk_id:string|null; sort_order:number; position_x:number|null; position_y:number|null; user_edited:boolean; mastery_score:number; mastery_status:'unassessed'|'weak'|'learning'|'mastered'; evidence_count:number; mastery_confidence:number }
export interface LearningEdge { id:string; from_node_id:string; to_node_id:string; relation_type:string; user_edited:boolean }
export interface KnowledgeNodeInput { name:string; node_type:string; description:string; exam_weight:number; estimated_minutes:number; position_x:number|null; position_y:number|null }
export interface KnowledgeEdgeInput { from_node_id:string; to_node_id:string; relation_type:string }
export interface KnowledgeArticle { id:string; node_id:string; title:string; body:string; source_chunk_id:string|null; sort_order:number; user_edited:boolean }
export interface LearningPlan { id:string; plan_date:string; title:string; total_minutes:number; generation_reason:string; created_at:string }
export interface PlanStage { id:string; plan_id:string; focus_node_id:string; title:string; description:string; status:'pending'|'active'|'complete'; estimated_days:number; sort_order:number }
export interface PlanNode { id:string; plan_id:string; node_id:string; task_type:string; title:string; estimated_minutes:number; sort_order:number }
export interface LearningAssets { knowledge_job:LearningAssetJob|null; plan_job:LearningAssetJob|null; knowledge_map:{nodes:LearningNode[];edges:LearningEdge[]}; handbook:{articles:KnowledgeArticle[]}; today_plan:{plan:LearningPlan|null;stages:PlanStage[];tasks:PlanNode[]} }
